src/features/expenses/AddExpense/MainInfo.tsx

```diff
@@ -0,0 +1,142 @@
+import currencies from "constants/currencies";
+import { ICreateTransaction } from "../types";
+import { newTransactionSelector } from "../slice";
+import { CreateRouteString, ExpenseRouteMode, composeExpenseRoute } from "constants/routes";
+import { saveTransaction } from "../slice";
+
+interface ILocationState {
+  proceed?: boolean;
+}
+
+const MainInfo = () => {
+  const { t } = useTranslation('expenses');
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const locationState = location.state as ILocationState;
+
+  const {groupId: groupIdOptional} = useParams();
+  const groupId = groupIdOptional ?? '';
+
+  const transaction = useSelector(newTransactionSelector);
+
+  const [form, setForm] = useState<ICreateTransaction>({
+    name: '',
+    paymentDateUtc: new Date().toISOString().split('T')[0],
+    currencyCode: currencies[0].value,
+    groupId: groupId,
+    payerDetails: [],
+    partialsAssignments: [],
+  });
+
+  const handleBackAction = useCallback(() => {
+    navigate(-1);
+  }, [navigate]);
+
+  const onControllTextChanged = useCallback((name: string, value: string) => {
+    setForm(form => ({
+      ...form,
+      [name]: value,
+    }));
+  }, []);
+
+  const onNextStep = () => {
+    dispatch(saveTransaction(form));
+    navigate(composeExpenseRoute(groupId, CreateRouteString, ExpenseRouteMode.ADD_MAININFO), {state: { proceed: true }});
+  }
+
+  useEffect(() => {  
+    if (!locationState?.proceed || !transaction) {
+      // Clean transaction
+      dispatch(saveTransaction(null));
+    }
+    else {
+      // Load transaction from the slice 
+      setForm((form) => ({
+        ...form,
+        ...transaction,
+      }));
+      return;
+    }
+  }, [dispatch, transaction, locationState])
+
+  return (
+    <PageWrapper>
+      <Header
+        text={t('mainInfo')}
+        leftActionComponent={<BackIconSVG onClick={handleBackAction} />}
+      />
+
+      <ContentWrapper fullWidth>
+        <MainInfoWrapper fullWidth>
+          <Column gap={'1rem'} fullWidth>
+            <MainInfoText>
+              {t('mainInfoText')}
+            </MainInfoText>
+
+            <Row jc={'center'} fullWidth>
+              <Input
+                reversedTheme
+                name={'name'}
+                displayName={t('transactionName')}
+                placeholder={t('transactionName')}
+                value={form.name}
+                onTextChange={onControllTextChanged}
+              />
+            </Row>
+            <Row jc={'center'} fullWidth>
+              <Input
+                reversedTheme
+                name={'paymentDateUtc'}
+                displayName={t('paymentDateUtc')}
+                type={'date'}
+                value={form.paymentDateUtc}
+                onTextChange={onControllTextChanged}
+              />
+            </Row>
+            <Row jc={'space-between'} ai={'center'} fullWidth>
+              <CurrencyText>
```

Comment:
It will be better to call it CurrencyTitle

Reply:
Done in: e3dfbabc6469a2498abbcb550c67b30a3716bba5